src/redux/thunks/updateFileEntry.ts

```diff
@@ -0,0 +1,96 @@
+import {createAsyncThunk, original} from '@reduxjs/toolkit';
+
+import fs from 'fs';
+import log from 'loglevel';
+import path from 'path';
+import {parse} from 'yaml';
+
+import {HELM_CHART_ENTRY_FILE, PREDEFINED_K8S_VERSION, ROOT_FILE_ENTRY} from '@constants/constants';
+
+import {RootState} from '@models/rootstate';
+
+import {UpdateFileEntryPayload} from '@redux/reducers/main';
+import {currentConfigSelector} from '@redux/selectors';
+import {getResourcesForPath} from '@redux/services/fileEntry';
+import {deleteResource, extractK8sResources, reprocessResources} from '@redux/services/resource';
+
+import {getFileStats, getFileTimestamp} from '@utils/files';
+
+export const updateFileEntry = createAsyncThunk(
+  'main/updateFileEntry',
+  async (payload: UpdateFileEntryPayload, thunkAPI: {getState: Function; dispatch: Function}) => {
+    const state: RootState = thunkAPI.getState();
+    const projectConfig = currentConfigSelector(state);
+    const schemaVersion = projectConfig.k8sVersion || PREDEFINED_K8S_VERSION;
+    const userDataDir = String(state.config.userDataDir);
+
+    try {
+      const fileEntry = state.main.fileMap[payload.path];
+      if (fileEntry) {
```

Comment:
I think that the logic from this file would be made a bit simple if we can exit early
`
if (!fileEntry) {log...}
`
and so of if it's possible